node/src/command.rs

```diff
@@ -28,6 +28,9 @@ pub fn run(version: VersionInfo) -> sc_cli::Result<()> {
     let chain_spec = args.chain.spec();
     let spec_factory = |_: &str| Ok(Box::new(chain_spec) as Box<_>);
 
+    let block_author = args.block_author();
+    let new_full_service = move |config| service::new_full(config, block_author);
```

Comment:
`suggestion`: 

For me, this could just be the following, since `block_author` is only useful for `new_full`. 

```suggestion
    let new_full_service = move |config| service::new_full(config, args.block_author());
```

Reply:
This does not work unfortunately. With your suggestion `args` will be moved to the closure and the closure will take ownership. However `args` is used later.

Reply:
Ah, right. You would need to `args.clone().block_author()`, I guess? yeah, let's not do that then.